test/fixture/locators.test.ts

```diff
@@ -173,5 +173,35 @@ test.describe('lib/fixture.ts (locators)', () => {
     })
   })
 
-  // TDOO: deferred page (do we need some alternative to `findBy*`?)
+  test.describe('deferred page', () => {
+    test.beforeEach(async ({page}) => {
+      await page.goto(`file://${path.join(__dirname, '../fixtures/late-page.html')}`)
+    })
+
+    test.afterEach(async ({page}) => page.close())
+
+    test('should handle the findBy* methods', async ({queries}) => {
+      const locator = await queries.findByText('Loaded!', undefined, {timeout: 7000})
+
+      expect(await locator.textContent()).toEqual('Loaded!')
+    })
+
+    test('should handle the findAllBy* methods', async ({queries}) => {
+      const locator = await queries.findAllByText(/Hello/, undefined, {timeout: 7000})
+
+      const text = await Promise.all([locator.nth(0).textContent(), locator.nth(1).textContent()])
+
+      expect(text).toEqual(['Hello h1', 'Hello h2'])
+    })
+
+    test('throws Testing Library error when locator times out', async ({queries}) => {
```

Comment:
1. Is there any reason why we called this `queries` and not [`screen`](https://testing-library.com/docs/queries/about/#screen)?
2. @gajus seemed to [expose the methods on `Page` itself](https://github.com/testing-library/playwright-testing-library/issues/430#issuecomment-1227650435). Would this be possible? Are we against doing this for some reason?

Reply:
Both things I have considered... I wasn't sure how much benefit we'd get from attaching stuff to `page`. From your experience, would this be convenient? We could attach `within` as well.

That said, I do like how switching to `screen` would resemble TL. Playwright's'`Page` is like a _real_ "screen," though.

Reply:
I quite like `screen` for familiarity with Testing Library. And, to be honest, I would prefer to not add any methods to `Page` as I think this would be unclear (e.g. "why do these methods exist when they're not on the Playwright website?!").

Reply:
We adopted the pattern I showed quite without regrets, for what it is worth.

I find that reducing verbosity in tests is extremely important to reduce test fatigue. These things add up

Reply:
Another idea -- could we leave `page` untouched but make `screen` itself be `Queries & Page`, so it is clear that you are using Testing Library but you do not need to juggle both?

That makes it independent but also reduces verbosity in tests.

Are there issues with doing this? What happens if there is a method that returns a `Page`? Is there anything we can do to avoid people needing to pass the `Locator`s we produce into `within(...)` in order to make the Testing Library queries available again?

In practice, being able to chain `.locator(...)` as you can do in Playwright is quite helpful.

Reply:
Ooo, I quite like that idea re: `screen` + `page`. I think I'll probably go that route.

As for your question, we could use a similar approach to augment the `Locator` returned from queries _with_ the query methods (so you could chain), but my concern there is departing from the Testing Library API.

Reply:
I'm think gonna address this stuff in a separate PR.